### PR TITLE
Refactor Grommet tests to remove react-test-renderer

### PR DESCRIPTION
**Pull request by Lukazovic**
#### What does this PR do?

Update Button legacy tests to use the standard render method from [react-testing-library](https://testing-library.com/docs/react-testing-library/intro/).

#### Where should the reviewer start?

`src/js/components/Grommet/__tests__/Grommet-test.js`

#### What testing has been done on this PR?

`yarn test`

#### How should this be manually tested?

Run: `yarn test`

#### What are the relevant issues?

#5197 - Testing - Refactor the usage of `renderer.create` to `render`

#### Do the grommet docs need to be updated?

No.

#### Should this PR be mentioned in the release notes?

No.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.

### CODE REVIEW

File: src/js/components/Grommet/__tests__/Grommet-test.js [hunk 1]
```diff
@@ -129,9 +126,8 @@ describe('Grommet', () => {
     /* eslint-enable max-len */
   ].forEach(ua => {
     test(`ssr rendering ${ua.substring(0, 25)}`, () => {
-      const component = renderer.create(<SSRTester ua={ua} />);
-      expect(component.toJSON()).toMatchSnapshot();
-      component.unmount();
+      const {container} = render(<SSRTester ua={ua} />);
+      expect(container.firstChild).toMatchSnapshot();
     });
```
Comment (jcfilben) [inline src/js/components/Grommet/__tests__/Grommet-test.js:131]: I think this test could be improved similar to what was done in the Responsive context tests, right now it is returning small for each option

Comment (Lukazovic) [reply src/js/components/Grommet/__tests__/Grommet-test.js:131]: Yeah, I agree. I'll try to improve it when I have some free time.

I even tried to run these tests with them wrapped in an `act()` and it returned "small" for each option as well. I think it is happening something similar to Responsive Context tests.
Screenshot from 2021-06-20 16-22-07

File: src/js/components/Grommet/__tests__/Grommet-test.js [hunk 2]
```diff
@@ -129,9 +122,11 @@ describe('Grommet', () => {
...
   ].forEach(ua => {
     test(`ssr rendering ${ua.substring(0, 25)}`, () => {
-      const component = renderer.create(<SSRTester ua={ua} />);
-      expect(component.toJSON()).toMatchSnapshot();
-      component.unmount();
+      act(() => {
+        const { container } = render(<SSRTester ua={ua} />);
+
+        expect(container.firstChild).toMatchSnapshot();
+      });
```
Comment (Lukazovic) [inline src/js/components/Grommet/__tests__/Grommet-test.js:129]: Hey, @jcfilben. I just improved the tests.

As I mentioned in the ResponsiveContext tests when we render with testing-library render it triggers the `useEffect` before starting the assertions. And as I also mentioned the `useEffect` changes the `size` based on the `document.body.clientWidth`. 

But in these specific tests, we want to test the `userAgent` so I decided to wrap the `render` in an `act` and test the snapshots inside this `act` so it doesn't trigger the `useEffect` which means we are testing the `userAgent`.

What do you think?

Comment (jcfilben) [reply src/js/components/Grommet/__tests__/Grommet-test.js:129]: Nice solution, looks good to me

**Review by jcfilben — APPROVED** [review]

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**